src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PropertyPages/DebugPageViewModel.cs

```diff
@@ -507,6 +444,209 @@ public ObservableList<NameValuePair> EnvironmentVariables
+            get
+            {
+                return LazyInitializer.EnsureInitialized(ref _removeEnvironmentVariableRowCommand, () =>
+                    new DelegateCommand(state =>
+                    {
+                        int oldIndex = EnvironmentVariablesRowSelectedIndex;
+                        EnvironmentVariables.RemoveAt(EnvironmentVariablesRowSelectedIndex);
+                        EnvironmentVariablesValid = true;
+                        EnvironmentVariablesRowSelectedIndex = (oldIndex == EnvironmentVariables.Count) ? oldIndex - 1 : oldIndex;
+                    }));
+            }
+        }
+                
+        public ICommand BrowseDirectoryCommand
+        {
+            get
+            {
+                return LazyInitializer.EnsureInitialized(ref _browseDirectoryCommand, () =>
+                    new DelegateCommand(state =>
+                    {
+                        using var dialog = new System.Windows.Forms.FolderBrowserDialog();
+                        string folder = WorkingDirectory;
+                        if (!string.IsNullOrEmpty(folder) && Directory.Exists(folder))
+                        {
+                            dialog.SelectedPath = folder;
+                        }
+                        DialogResult result = dialog.ShowDialog();
+                        if (result == DialogResult.OK)
+                        {
+                            WorkingDirectory = dialog.SelectedPath;
+                        }
+                    }));
+            }
+        }
+                
+        public ICommand BrowseExecutableCommand
+        {
+            get
+            {
+                return LazyInitializer.EnsureInitialized(ref _browseExecutableCommand, () =>
+                    new DelegateCommand(state =>
+                    {
+                        using var dialog = new System.Windows.Forms.OpenFileDialog();
+                        string file = ExecutablePath;
+                        if (!string.IsNullOrEmpty(file) && (file.IndexOfAny(Path.GetInvalidPathChars()) == -1) && Path.IsPathRooted(file))
+                        {
+                            dialog.InitialDirectory = Path.GetDirectoryName(file);
+                            dialog.FileName = file;
+                        }
+                        dialog.Multiselect = false;
+                        dialog.Filter = _executableFilter;
+                        DialogResult result = dialog.ShowDialog();
+                        if (result == DialogResult.OK)
+                        {
+                            ExecutablePath = dialog.FileName;
+                        }
+                    }));
+            }
+        }
+
+        public bool NewProfileEnabled => LaunchProfiles != null;
+
+        public ICommand NewProfileCommand
+        {
+            get
+            {
+                return LazyInitializer.EnsureInitialized(ref _newProfileCommand, () =>
+                 new DelegateCommand(state =>
```

Comment:
Indentation looks funny here (it was like this beforehand though).

Reply:
Yeah might fix it latter.